src/core/mesh/qgsmeshlayerrenderer.cpp

```diff
@@ -360,3 +363,28 @@ void QgsMeshLayerRenderer::renderVectorDataset()
 
   renderer.draw();
 }
+
+void QgsMeshLayerRenderer::renderVectorTrace()
```

Comment:
i am not sure if vector trace should be new renderer class or part of `QgsMeshVectorRenderer` Probably depends how much code of `QgsMeshVectorRenderer` can be used in trace rendering. I was maybe thinking if the trace renderer is part of `QgsMeshVectorRenderer` but the calculation of streamlines is separate class (in analysis library maybe?). Maybe you have some insight/feeling about which structure is better?

Reply:
Right now, I am not sure... Maybe, it would depend how the GUI would be implemented. I was thinking about doing another tab in the properties of the layer for streamlines options because those option will maybe different from vector options...